### ISSUE TITLE
Docker Desktop 4.3.2 GELF Logging Driver not working (again)

### ISSUE THREAD

**Issue by lofbacke** [docker/for-win#12465]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: D546E9F4-8C1F-48CE-A431-189457BE23FB/20211224190428

### Actual behavior
Docker container does not forward logs to designated UDP port using GELF logging driver.

### Expected behavior
Docker container should forward logs from the container to the UDP port designated in the docker-compose file.

### Information
This is a repeat issue that was previously reported (and resolved) in issue report [#10081](https://github.com/docker/for-win/issues/10081). It is now back after updating from Docker Desktop 3.6.0 to 4.3.2 (yes, I held off upgrading to version 4 for a few months). All of the symptoms are identical.

  - Windows Version: Windows 11 Home Edition 21H2, 22000.376
  - Docker Desktop Version: 4.3.2 (72729)
  - WSL2 or Hyper-V backend? WSL2
  - Are you running inside a virtualized Windows e.g. on a cloud server or a VM: No

### Output of `& "C:\Program Files\Docker\Docker\resources\com.docker.diagnose.exe" check`
```
[358:14:16:53.103][I] ipc.NewClient: d6f6df7b-com.docker.diagnose -> \\.\pipe\dockerBackendV2 com.docker.service.exe
[goroutine 1 [running, locked to thread]:
[win/backend/pkg/service.NewClientForPath(...)
[       win/backend/pkg/service/service.go:26
[win/backend/pkg/service.NewClient(...)
[       win/backend/pkg/service/service.go:20
[common/pkg/diagkit/gather/diagnose.init()
[       common/pkg/diagkit/gather/diagnose/ipc_windows.go:23 +0x13b
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[SKIP] DD0018: does the host support virtualization?
[PASS] DD0002: does the bootloader have virtualization enabled?
[PASS] DD0017: can a VM be started?
[PASS] DD0024: is WSL installed?
[PASS] DD0021: is the WSL 2 Windows Feature enabled?
[PASS] DD0022: is the Virtual Machine Platform Windows Feature enabled?
[PASS] DD0025: are WSL distros installed?
[PASS] DD0026: is the WSL LxssManager service running?
[PASS] DD0029: is the WSL 2 Linux filesystem corrupt?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0005: is the user in the docker-users group?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0006: is the Docker Desktop Service responding?
[PASS] DD0012: is the VM networking working?
[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0033: does the host have Internet access?
No fatal errors detected.
```

### Steps to reproduce the behavior
The missing log entries can easily be replicated using the following steps, originally described in the previous issue report in [this post](https://github.com/docker/for-win/issues/10081#issuecomment-797184665).
  1. Create a temporary directory anywhere on your system
  2. Add a `docker-compose.yml` file with the following contents:
```
version: '3.5'

services:
  seq-input-gelf:
    image: datalust/seq-input-gelf:latest
    depends_on:
      - seq
    ports:
      - "12201:12201/udp"
    environment:
      SEQ_ADDRESS: "http://seq:5341"
    restart: unless-stopped

  seq:
    image: datalust/seq:latest
    ports:
      - "5341:80"
    environment:
      ACCEPT_EULA: "Y"
    restart: unless-stopped
    volumes:
      - ./seq-data:/data

  api:
    image: bash:latest
    depends_on:
      - seq-input-gelf
    volumes:
      - ./script.sh:/script.sh:ro
    command: bash /script.sh
    logging:
      driver: gelf
      options:
        gelf-address: "udp://localhost:12201"
```
  3. Add a `script.sh` file with the following contents (ensure that the file uses UNIX line endings):
```
sleep 5
for i in 1 2 3 4 5
do
   sleep 2
   echo "Test log number $i"
done
```
  4. Run `docker-compose up` from the directory
  5. Compare the output from the container with the output of `docker logs <container ID>`. The result of the `docker logs` command is clearly missing log entries that the container emitted.
![image](https://user-images.githubusercontent.com/52764594/147370148-a173008e-3f7a-465c-b703-d39594b890f6.png)

**Comment by thaJeztah** [comment]
odd

random thought; does it make a difference if you use `"udp://127.0.0.1:12201"` (IPv4 IP-address instead of `localhost`) ?

**Comment by lofbacke** [comment]
@thaJeztah Actually, I just tried that and yes, it fixes the issue! Thanks for the tip. It used to work flawlessly with `localhost`. Is this something that can be resolved in the future?

**Comment by thaJeztah** [comment]
Hmm, right, so what I suspect is that `localhost` is resolved to both the IPv4 (`127.0.0.1`) and IPv6 (`::1`) address, and that the logs are round-robin sent to those.

I seem to recall (but I'm not on a Windows machine) that IPv6 didn't work properly in WSL2, which may be related to this.

So, yes, using the IP address can be used as a workaround, but it probably should be looked into.

Mostly out of curiosity; if you use the IPv6 address, do any logs arrive at all, or will it result in 100% loss? (and probably needs square brackets around the IP address to make it work combined with the `:<port>`, so `"udp://[::1]:12201"`)

**Comment by lofbacke** [comment]
@thaJeztah When I use `udp://[::1]:12201`, I get the exact same results as screen-shotted in my original post. Seq gets nothing, and the `docker logs` command shows output from log numbers 1, 3, and 5.
